# Assignments

1. If you already have a DMI-TCAT data collection running, download your Twitter data (be careful with the amount of tweets - if you have more than 200,000, get only a subsample by selecting less days). If you don't have data yet, use it from another person.

2. Create a new Jupyter notebook and load your data using pandas

3. Do some basic descriptives to check:
    * How many tweets are there?
    * What is the average number of friends and of followers that users tweeting about your topic have?
    * What is the average number of retweets that each tweet got?
    * How many tweets are there per language?

4. Select a sample of tweets on which you'd like to run sentiment analysis and export just the id and the text of the tweet to a Pickle (.pkl) file. We can do sentiment analysis in up to 24 languages, but only one language at a time. So if you need two languages, you need to save one file per language.
    * **Important:** The files need to have your name and the language in the filename (two-letter ISO code, upper case). In my case, the files would be called: "TheoAraujo_EN.pkl", "TheoAraujo_PT.pkl". Upload the file(s) to our dropbox using the link available at our General Repository (see DA2 - end of the page).
    * To do this, you'll need to find out how to do a few things yourself. Check out in Google or in online tutorials how to:
        1. Checking which columns a pandas dataframe has 
        2. Filter a Pandas dataframe based on values in one column
        3. Select only certain columns of a dataframe in Pandas

In [2]:
import pandas as pd

In [3]:
tweets = pd.read_pickle('JingjingWang_export.pkl')

In [4]:
tweets

id          created_at   from_user_name  \
0       1093857236068302849 2019-02-08 13:01:02         GriffBed   
1       1093857235934068737 2019-02-08 13:01:02         AppLovin   
2       1093857238354198528 2019-02-08 13:01:03   NewsMobilidade   
3       1093857240887541760 2019-02-08 13:01:03     gabriel_irdk   
4       1093857248919633920 2019-02-08 13:01:05      cmccaerdydd   
5       1093857259023683585 2019-02-08 13:01:08            laiki   
6       1093857261146001408 2019-02-08 13:01:08      DKaiserREAL   
7       1093857263373221888 2019-02-08 13:01:09     _busnescymru   
8       1093857263658430464 2019-02-08 13:01:09        AirTunnel   
9       1093857265688473600 2019-02-08 13:01:09        bobbo8181   
10      1093857268020506626 2019-02-08 13:01:10  RomuloLamblet13   
11      1093857270075674624 2019-02-08 13:01:10         ClanMpex   
12      1093857270855811072 2019-02-08 13:01:10    PatriotPennsy   
13      1093857271522750464 2019-02-08 13:01:11     JullyRochaII   
14      1093857272026066944 2019-02-08 13:01:11  blackmediaskate   
15      1093857279177342976 2019-02-08 13:01:12           Be_Sod   
16      1093857280012021760 2019-02-08 13:01:13    flordase_naty   
17      1093857288367099904 2019-02-08 13:01:15  james_a_timmins   
18      1093857299653971968 2019-02-08 13:01:17       techristic   
19      1093857300727648256 2019-02-08 13:01:18       hot21radio   
20      1093857313725779968 2019-02-08 13:01:21           Caequi   
21      1093857319727833088 2019-02-08 13:01:22          HwbNews   
22      1093857327566983168 2019-02-08 13:01:24        aamorizer   
23      1093857332436570113 2019-02-08 13:01:25  CsarAugustoSa11   
24      1093857334315700224 2019-02-08 13:01:26    baletomaanija   
25      1093857335141777408 2019-02-08 13:01:26       aerthonnta   
26      1093857341022371840 2019-02-08 13:01:27       aerthonnta   
27      1093857341215305729 2019-02-08 13:01:27       sahsahesco   
28      1093857341945143296 2019-02-08 13:01:27        qlfoivick   
29      1093857350388236289 2019-02-08 13:01:29    RosendoMateus   
...                     ...                 ...              ...   
237717  1095278486762119168 2019-02-12 11:08:35           vvvmpn   
237718  1095278492793503744 2019-02-12 11:08:36       HYamaguchi   
237719  1095278495347949569 2019-02-12 11:08:37           iRaxy1   
237720  1095278512104267776 2019-02-12 11:08:41     AndreiwKim21   
237721  1095278517120630784 2019-02-12 11:08:42   Thaiis_plicado   
237722  1095278519502950400 2019-02-12 11:08:42        elbritobr   
237723  1095278533780201473 2019-02-12 11:08:46      lismaismail   
237724  1095278535109943297 2019-02-12 11:08:46  nildaalcarinque   
237725  1095278538838675456 2019-02-12 11:08:47     DreemurrKris   
237726  1095278545973243904 2019-02-12 11:08:49        xiuvoces_   
237727  1095278563442470912 2019-02-12 11:08:53     Zete81323679   
237728  1095278570476253185 2019-02-12 11:08:55         yayy_kfc   
237729  1095278570610573312 2019-02-12 11:08:55      Dangonzaga6   
237730  1095278576167993345 2019-02-12 11:08:56     Rajneesh0707   
237731  1095278576008478720 2019-02-12 11:08:56       AdamKasim_   
237732  1095278578088976384 2019-02-12 11:08:56        srmarilia   
237733  1095278578021761024 2019-02-12 11:08:56   Shitao42300484   
237734  1095278580739772417 2019-02-12 11:08:57       Yankaronin   
237735  1095278581402517505 2019-02-12 11:08:57    ihimanshusahu   
237736  1095278582610325505 2019-02-12 11:08:58       love_061_3   
237737  1095278583893889024 2019-02-12 11:08:58          Mobilly   
237738  1095278584795676674 2019-02-12 11:08:58      thamizoando   
237739  1095278586968358912 2019-02-12 11:08:59     NatPennArBed   
237740  1095278595138703360 2019-02-12 11:09:01   leinsecuregirl   
237741  1095278598162993152 2019-02-12 11:09:01     sandraiansen   
237742  1095278606694182918 2019-02-12 11:09:03       LikeImHalf   
237743  1095278608673841152 2019-02-12 11:09:04       nickinglis   
237744  10952

**How many tweets are there?**

In [4]:
tweets.describe().transpose()

count          mean           std  \
from_user_id                237747.0  4.046125e+17  4.694585e+17   
from_user_tweetcount        237747.0  4.020305e+04  1.350499e+05   
from_user_followercount     237747.0  6.632670e+03  1.404806e+05   
from_user_friendcount       237747.0  1.317601e+03  1.124337e+04   
from_user_listed            237747.0  6.579501e+01  8.274142e+02   
from_user_verified          237747.0  1.385927e-02  1.169070e-01   
from_user_favourites_count  237747.0  1.525176e+04  3.536276e+04   
geo_lat                       1538.0  2.608411e+01  1.744078e+01   
geo_lng                       1538.0 -2.777624e+01  6.790758e+01   
retweet_id                  126898.0  1.092701e+18  2.222832e+16   
retweet_count               237747.0  0.000000e+00  0.000000e+00   
favorite_count              237747.0  0.000000e+00  0.000000e+00   
to_user_id                   28594.0  3.198022e+17  4.479300e+17   
in_reply_to_status_id        27018.0  1.093942e+18  1.480761e+16   
possibly_sensitive           64332.0  2.998508e-02  1.705474e-01   
quoted_status_id             13805.0  1.092935e+18  1.833869e+16   

                                     min           25%           50%  \
from_user_id                5.720000e+02  4.220787e+08  3.244284e+09   
from_user_tweetcount        1.000000e+00  2.047000e+03  9.450000e+03   
from_user_followercount     0.000000e+00  1.030000e+02  3.630000e+02   
from_user_friendcount       0.000000e+00  1.440000e+02  3.630000e+02   
from_user_listed            0.000000e+00  0.000000e+00  1.000000e+00   
from_user_verified          0.000000e+00  0.000000e+00  0.000000e+00   
from_user_favourites_count  0.000000e+00  5.850000e+02  3.916000e+03   
geo_lat                    -5.164768e+01  2.474671e+01  3.375545e+01   
geo_lng                    -1.578218e+02 -9.228960e+01 -4.800744e+01   
retweet_id                  1.216394e+09  1.093978e+18  1.094639e+18   
retweet_count               0.000000e+00  0.000000e+00  0.000000e+00   
favorite_count              0.000000e+00  0.000000e+00  0.000000e+00   
to_user_id                  1.186000e+03  1.951094e+08  2.253240e+09   
in_reply_to_status_id       4.990441e+09  1.094073e+18  1.094531e+18   
possibly_sensitive          0.000000e+00  0.000000e+00  0.000000e+00   
quoted_status_id            2.962480e+17  1.093922e+18  1.094604e+18   

                                     75%           max  
from_user_id                9.221864e+17  1.095275e+18  
from_user_tweetcount        3.304000e+04  9.585542e+06  
from_user_followercount     1.137000e+03  1.637227e+07  
from_user_friendcount       8.940000e+02  1.209471e+06  
from_user_listed            1.100000e+01  1.283850e+05  
from_user_verified          0.000000e+00  1.000000e+00  
from_user_favourites_count  1.475550e+04  1.331066e+06  
geo_lat                     3.527842e+01  5.695640e+01  
geo_lng                     4.664203e+01  1.470099e+02  
retweet_id                  1.094992e+18  1.095278e+18  
retweet_count               0.000000e+00  0.000000e+00  
favorite_count              0.000000e+00  0.000000e+00  
to_user_id                  8.398158e+17  1.095221e+18  
in_reply_to_status_id       1.094920e+18  1.095278e+18  
possibly_sensitive          0.000000e+00  1.000000e+00  
quoted_status_id            1.094878e+18  1.095277e+18

There are 237747 tweets in total

**What is the average number of friends and of followers that users tweeting about your topic have?**

In [16]:
tweets['from_user_followercount'].mean()

6632.669972702074

In [18]:
tweets['from_user_friendcount'].mean()

1317.6007604722668

**What is the average number of retweets that each tweet got?**

In [7]:
retweets = pd.read_pickle('JingjingWang_retweets.pkl')

In [8]:
retweets.describe()

retweets_total
count    17516.000000
mean         7.242978
std         69.150830
min          1.000000
25%          1.000000
50%          1.000000
75%          3.000000
max       5424.000000

The average number of tweets is 7.24

**How many tweets are there per language?**

In [9]:
tweets.groupby('lang').describe()

favorite_count                                     \
              count mean  std  min  25%  50%  75%  max   
lang                                                     
ar            736.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
bg              6.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
bn              5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ca           1123.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
cs             43.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
cy           6042.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
da             96.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
de            859.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
el              5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
en          71534.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
es           8408.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
et            267.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
eu             46.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
fa              5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
fi            193.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
fr           2567.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
gu              6.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
hi            875.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ht            187.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
hu             66.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
in           6669.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
is             58.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
it           1061.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
iw              2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ja           6815.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ko           2303.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
lt            793.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
lv           7384.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ml              2.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
mr             19.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
my              1.0  0.0  NaN  0.0  0.0  0.0  0.0  0.0   
nl            382.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
no            138.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
pl            126.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
pt          99568.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ro           1751.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ru            136.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
sl             27.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
sv            357.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ta             19.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
th           7329.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
tl           1138.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
tr           5353.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
uk              6.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
und          3078.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
ur              7.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
vi             54.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
zh            102.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

     from_user_favourites_count                    ...         retweet_id  \
                          count          mean      ...                75%   
lang                                               ...                      
ar                        736.0   4468.733696      ...       1.094513e+18   
bg                          6.0   4523.000000      ...       1.095199e+18   
bn                          5.0    150.600000      ...       1.094159e+18   
ca                       1123.0   8278.991095      ...       1.094275e+18   
cs                         43.0  10243.511628      ...       1.093771e+18   
cy                       6042.0  11308.210526      ...       1.094881e+18   
da                         96.0  17458.927083      ...       1.094498e+18   
de                        859.0  13046.756694      ...       1.094921e+18   
el                          5.0    142.600000      ...       1.094673e+18   
en                      71534.0  16176.469972      ...       1.094840e+18   
es                       8408.0  16932.718720

**Sentiment analysis data cleaning**

In [10]:
tweets.columns

Index(['id', 'created_at', 'from_user_name', 'from_user_id', 'from_user_lang',
       'from_user_tweetcount', 'from_user_followercount',
       'from_user_friendcount', 'from_user_listed', 'from_user_realname',
       'from_user_utcoffset', 'from_user_timezone', 'from_user_description',
       'from_user_url', 'from_user_verified', 'from_user_profile_image_url',
       'from_user_created_at', 'from_user_withheld_scope',
       'from_user_favourites_count', 'source', 'location', 'geo_lat',
       'geo_lng', 'text', 'retweet_id', 'retweet_count', 'favorite_count',
       'to_user_id', 'to_user_name', 'in_reply_to_status_id', 'filter_level',
       'lang', 'possibly_sensitive', 'quoted_status_id', 'withheld_copyright',
       'withheld_scope'],
      dtype='object')

In [11]:
english = tweets[tweets['lang'] == 'en']

In [12]:
english[['id', 'text', 'lang']]

id  \
1       1093857235934068737   
8       1093857263658430464   
9       1093857265688473600   
11      1093857270075674624   
12      1093857270855811072   
17      1093857288367099904   
18      1093857299653971968   
19      1093857300727648256   
30      1093857353567559684   
37      1093857395762188293   
38      1093857401747300352   
43      1093857426443513856   
47      1093857441240965120   
51      1093857456537653252   
52      1093857461826596870   
57      1093857504331628545   
58      1093857518898610176   
65      1093857556945125376   
66      1093857557041631233   
69      1093857582257786880   
74      1093857620006453253   
79      1093857640827011072   
82      1093857662662598656   
89      1093857704987156480   
94      1093857732522856448   
96      1093857757965496320   
103     1093857785224331266   
105     1093857793193533442   
106     1093857793965080577   
108     1093857810776031233   
...                     ...   
237644  1095278166115987456   
237649  1095278181332840448   
237652  1095278191051190273   
237654  1095278193240608768   
237655  1095278193395814401   
237660  1095278226912403456   
237664  1095278238291685376   
237666  1095278241462583296   
237667  1095278247959396355   
237669  1095278254879981570   
237674  1095278269803450368   
237684  1095278333871280129   
237690  1095278355564412929   
237693  1095278358227836929   
237695  1095278364284387328   
237697  1095278366826090496   
237705  1095278457104285696   
237707  1095278459688009728   
237710  1095278464482099205   
237714  1095278471071199232   
237715  1095278475684896769   
237716  1095278477346029568   
237720  1095278512104267776   
237723  1095278533780201473   
237725  1095278538838675456   
237730  1095278576167993345   
237735  1095278581402517505   
237743  1095278608673841152   
237745  1095278612591259648   
237746  1095278610758361089   

                                                     text lang  
1       .@lewisleong on why the Oculus Go and Oculus Q...   en  
8       AR.Drone Parrot 2.0 Elite Edition in Sand with...   en  
9       RT @heyitsCarolyn: Watch this little snippet f...   en  
11      RT @h1z1hunt: 💰$500 GIVEAWAY!💰\n\n🎁3x AR PATRI...   en  
12      RT @heyitsCarolyn: Watch this little snippet f...   en  
17      RT @Purcelluk: Yesterday morning, our Cambridg...   en  
18      How Lush is using AR to change the way you buy...   en  
19      #NowPlaying : Conversation by AR Williams #lis...   en  
30      RT @claudiomkd: RT Ronald_vanLoon: This app is...   en  
37      RT @heyitsCarolyn: Watch this little snippet f...   en  
38      @maydaymindy9 @PersianKittenz Ar tge end of th...   en  
43      RT @claudiomkd: RT Ronald_vanLoon: This app is...   en  
47      RT @kashthefuturist: This is creepy\n\n#Augmen...   en  
51      AR = Who Pays For Ad Space?: Imagine, if you w...   en  
52      AR (SHARP): latest activity (24h) (server2) #s...   en  
57      RT @_SamKeene: Wow! Thanks for illustrating my...   en  
58      RT @thenextweb: How Lush is using AR to change...   en  
65      How Lush is using AR to change the way you buy...   en  
66      Click to see how easy it is to #augment your b...   en  
69      @KaitMarieox Have a bunch of kids and buy them...   en  
74      @ArchipalagoTV Woho fuuu s3 boliga kramaen!! K...   en  
79      RT @GOGLINJF: Experience #Google Earth in #Vir...   en  
82      @Daisy84x Thanks very much Daisy, that's all r...   en  
89      OTV, ORTEL, MS TELECOM and Calorex holdings, a...   en  
94      RT @kashthefuturist: This is creepy\n\n#Augmen...   en  
96      @lnVayn Matt still , I get it he would be amaz...   en  
103     RT @heyitsCarolyn: Watch this little snippet f...   en  
105     @ArchipalagoTV Woho fuuu s3 boliga kramaen!! K...   en  
106     @ARMurugadoss @ivenkatmohan @VishalKOfficial @...   en  
108     RT @STEPHMHAMILL: My Parents gave me an early ...   en  
...                                                   ...  ...  
237644  RT @squintneon

In [13]:
sentiment_analysis_sample = english[['id', 'text']]

In [14]:
sentiment_analysis_sample.to_pickle('JingjingWang_EN.pkl')

In [15]:
chinese = tweets[tweets['lang'] == 'zh']

In [16]:
chinese[['id', 'text', 'lang']]

id  \
78      1093857639904104448   
5045    1093881349612670976   
6594    1093887814972960769   
24289   1093972856650649600   
24348   1093973139392851968   
24621   1093974605922852864   
32748   1094023731477917701   
34879   1094038883174662145   
36856   1094053905447702533   
38823   1094068996519927808   
40942   1094087177145851905   
41024   1094087732450783233   
42133   1094099202689429504   
43503   1094114273582817280   
43815   1094117916537245696   
44751   1094129447350034433   
46067   1094144559049834496   
47485   1094159570480648192   
49000   1094174683644260352   
50579   1094189822715936769   
52469   1094204896180613120   
54627   1094220028814245888   
59156   1094250249110126592   
59758   1094254364762521600   
68080   1094307410494205952   
70485   1094321674902630400   
79501   1094401182607523841   
81048   1094416313576054785   
81127   1094417078684135424   
83163   1094431395617914880   
...                     ...   
151102  1094912319493029890   
151491  1094914587109908480   
151493  1094914601374801920   
151968  1094917084016914432   
154230  1094929708150546433   
156905  1094944594737930240   
156944  1094944777345196033   
158896  1094954748359462914   
159666  1094958352638054402   
160011  1094959797990973440   
162290  1094971582655619073   
162438  1094972340100689920   
162991  1094974970667487232   
163536  1094977674194300928   
164202  1094980992882872320   
164910  1094984166741966851   
165041  1094984700869910528   
178800  1095020535820382208   
204822  1095108928944320512   
207830  1095117543927209984   
208455  1095119453816348672   
208459  1095119464557879298   
208472  1095119501811744769   
209000  1095121292032827392   
212247  1095131417275654144   
221504  1095167055681081344   
222143  1095170662723977216   
225639  1095200123695583232   
225641  1095200138954526720   
232883  1095253502085992448   

                                                     text lang  
78      新春結伴拜年 「細龍太」與MIRROR成員關係曝光 #MSNHongKong #MSN香港 ...   zh  
5045    年初四 🏮🏮 Heng ar.. Huat ar.. Hea ar.. \n周末来临又可以穿...   zh  
6594    15大正妹合成！「韓國終極美女」照片出爐竟撞臉她 #MSNHongKong #MSN香港 h...   zh  
24289   2019 年初 AR 趨勢摘要 (quoting Digi-Capital) https:/...   zh  
24348   2019 年初 AR 趨勢摘要 (quoting Digi-Capital) https:/...   zh  
24621   2019 年初 AR 趨勢摘要 (quoting Digi-Capital) https:/...   zh  
32748   張家輝13歲囡囡張童 亭亭玉立似關詠荷 #MSNHongKong #MSN香港 https:...   zh  
34879   撐善舉受傷向家人致歉 麥明詩望豬年有拖拍 #MSNHongKong #MSN香港 https...   zh  
36856   張智霖袁詠儀結婚18年雪地放閃 偶遇同日結婚鄧超孫儷 #MSNHongKong #MSN香港...   zh  
38823   網友插黃心穎返鄉下拜年勁高傲 馬國明急住護航解話！ #MSNHongKong #MSN香港 ...   zh  
40942   春晚高科技：5G+4K，AR&amp;VR，你怕了吗？\n\nhttps://t.co/dn...   zh  
41024   春晚高科技：5G+4K，AR&amp;VR，你怕了吗？ https://t.co/ch77o...   zh  
42133   美斯遠拋C朗成最高薪球星 #MSNHongKong #MSN香港 https://t.co/...   zh  
43503   《廉政風雲》落重本 青雲張家輝「回春」2分鐘耗資千萬 #MSNHongKong #MSN香港...   zh  
43815   要開工了好鬱卒嗎😔😔從2/9開始每個週末早晨#造型氣球甜心-Eva與您一起開啟美好的週末🎈🎈...   zh  
44751   【結婚前奏】懶理吳千語未忘情 林峯陪張馨月回鄉過年 #MSNHongKong #MSN香港 ...   zh  
46067   結婚唔係一時衝動 陳曉東：幾年前開始追求「回家的感覺」 #MSNHongKong #MSN香...   zh  
47485   家人無催拍拖 徐子珊: 已經咁多年 應該放棄咗 #MSNHongKong #MSN香港 ht...   zh  
49000   盤點8個韓星有效成功瘦身秘訣 節日過後要減啦！ #MSNHongKong #MSN香港 ht...   zh  
50579   溝女無數嗌破產 林作貼街招＄100教英文：我要錢 #MSNHongKong #MSN香港 h...   zh  
52469   睇《新喜劇之王》憶咖哩啡歲月 杜汶澤感觸落淚 #MSNHongKong #MSN香港 htt...   zh  
54627   家人放棄催婚 李佳芯豬年冇結婚生仔計劃 #MSNHongKong #MSN香港 https:...   zh  
59156   家人催生豬B 胡琳自爆婚禮拋「花女花球」 #MSNHongKong #MSN香港 https...   zh  
59758   Siu Fa's elder brother, Ar Fook!! He loves hoo...   zh  
68080                          @Pennyho9 安美露好work ar 鬆左好多   zh  
70485   【数千发弹药和武器！加州警方在帮派成员家中发现秘密地下射击场】加州警察局发现了一个秘密的地下...   zh  
79501   關心妍囡囡向許廷鏗苟姑娘逗利是 網民大讚可愛 #MSNHongKong #MSN香港 htt...   zh  
81048   14歲細女黐身攬實 Fit爆彭羚放真‧林狗 #MSNHongKong #MSN香港 http...   zh  
81127            极客漫画：增强现实（AR）2.0 https://t.co/2KoFJxv9AW   zh  
83163   素顏周汶錡晒與大仔盞鬼賣萌照 網友：皮膚幾好！ #MSNHongKong #MSN香港 ht...   zh  
...                                                   ...  ...  
151102  谷歌地图开始让部分用户测试新

In [17]:
chinese_sample = english[['id', 'text']]

In [18]:
chinese_sample.to_pickle('JingjingWang_ZH.pkl')

**Mission Completed** :)